.github/workflows/build_test.yml

```diff
@@ -126,20 +149,47 @@ jobs:
           name: windows_py${{ matrix.python-version }}
           path: dist/
 
+  test-sdist:
+    needs: build-sdist
+    name: Test sdist
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v1
+      - name: Download sdist
+        uses: actions/download-artifact@v2
+        with:
+          name: sdist
+          path: dist
+      - name: Show sdist downloaded
+        run: ls -lh dist
+      - name: Set up Python
+        uses: actions/setup-python@v2
+        with:
+          python-version: '3.6'
+      - name: Display Python version
+        run: python -c "import sys; print(sys.version)"
+      - name: Upgrade setuptools and pip
+        run: python -m pip install -U setuptools pip
+      - name: Install sdist
+        run: pip install dist/*
+
   test:
     name: Test ${{ matrix.os.download_name }}-${{ matrix.python-version }}-${{ matrix.cloud-provider }}
-    needs: [build-manylinux, build-macos, build-windows]
+    needs: ${{ matrix.os.needs }}
```

Comment:
Unfortunately it seems like this is not possible.
See: https://github.community/t/matrix-variable-does-not-work-in-needs-keyword/16972/2
Going to roll this back.